server/s3_get_object_action.cc

```diff
@@ -494,6 +494,17 @@ void S3GetObjectAction::send_data_to_client() {
     LOG_PERF("get_object_send_data_ms", request_id.c_str(), mss);
     s3_stats_timing("get_object_send_data", mss);
 
+    std::string s_md5_calc = motr_reader->get_content_md5_base64();
+    std::string s_md5_read = object_metadata->get_md5();
+    s3_log(S3_LOG_ERROR, request_id, "MD5 calculated: %s, MD5 read %s", /* XXX */
+           s_md5_calc.c_str(), s_md5_read.c_str());     /* XXX */
+    s3_log(S3_LOG_DEBUG, request_id, "MD5 calculated: %s, MD5 read %s",
+           s_md5_calc.c_str(), s_md5_read.c_str());
+
+    if (s_md5_calc != s_md5_read) {
+      s3_log(S3_LOG_ERROR, request_id, "Content MD5 mismatch\n");
+      set_s3_error("BadDigest");
```

Comment:
Better if error is set to "InternalError"
BadDigest error is used if s3 client has provided MD5 of the file and there is mismatch, here s3 client didn't provide any md5 of the file

Reply:
Done.